Tribler/community/tunnel/tunnel_community.py

```diff
@@ -466,9 +466,9 @@ def create_circuit(self, goal_hops, ctype=CIRCUIT_TYPE_DATA, callback=None, max_
         if not required_endpoint:
             for c in self.dispersy_yield_verified_candidates():
                 pubkey = c.get_member().public_key
+                exit_candidate = self.exit_candidates[pubkey]
```

Comment:
Yep. They're managed by overriding the callback for the dispersy-introduction-request and dispersy-introduction-response messages in the TunnelCommunity, and cleaned up periodically by the do_remove thing.